### ISSUE TITLE
Websocket chokes on refused connection

### ISSUE THREAD

**Issue by timokoch** [VirtualProgrammingLab/viplab-websocket-api#13]
I'm running a viplab front end locally, setting up the necessary services with the following docker-compose setup:

```
version: "3"

services:
  activemq:
    image: viplab-artemis:2.11.0
    build:
      context: ./artemis
      args:
        ACTIVEMQ_ARTEMIS_VERSION: "2.11.0"
    ports:
      - 8084:8161
      - 5672:5672
    environment:
      ARTEMIS_USERNAME: user
      ARTEMIS_PASSWORD: password
  websocket-api:
    build: ./websocket-api
    ports:
      - 8083:8080
    environment:
      JWKS_FILE: /config/jwks.json
    volumes:
      - ./jwks.json:/config/jwks.json:ro
    depends_on:
      - activemq
  standalone-frontend:
    image: viplab/standalone-frontend:latest
    ports:
      - 8085:5000
    volumes:
      - ./jwks.private.json:/jwks.private.json:ro
      - ./examples:/input:ro
```

Sometimes the services start-up fine and sometimes the websocket fails to connect to activemq and dies with an error: `java.net.ConnectException: Connection refused`. I assume the activemq is not ready yet in these cases. I attached the startup log:

[startup.log](https://github.com/VirtualProgrammingLab/viplab-websocket-api/files/4614491/startup.log)

__Observed behaviour__:
The websocket dies.

__Expected behaviour__:
The websocket tries again later to establish the connection.

**Comment by Legion2** [comment]
Simples solution is to ensure activemq is running before starting websocket-api, because we have no control over the initial client connection retries provided be the quarkus amqp library.

**Comment by timokoch** [comment]
A workaround could be a wrapper script around the websocket startup, see https://docs.docker.com/compose/startup-order/

In order to do that I would need a safe way to query if the activemq is ready. Is it enough to check if the port 5672 accepts TCP connection? or can you/someone suggest a better way to write an application specific check (second option in the link)?

**Comment by timokoch** [comment]
Does the retry decorator work or is that not applicable for initial connections? https://quarkus.io/guides/microprofile-fault-tolerance

**Comment by Legion2** [comment]
The decorator don't work, because there is no method to be annotated, the amqp client is created in the background by quarkus

**Comment by Legion2** [comment]
We should open an issue at quarkus, so that either qurakus exits if there is an error with the stream processing at startup or the connections to message brokers are retried and don't leave the appliction in a broken state.

**Comment by pascalseeland** [comment]
There is already on open issue at quarkus on this topic: https://github.com/quarkusio/quarkus/issues/7404

I did add a dependency from the websocket-api to activemq on the https://github.com/VirtualProgrammingLab/viplab-docker-dev-env/commit/6267ce49e5e6f39498c874f2576ed1bfebe82303  At least on my system it did help that quarkus is at least started after artemis. Maybe this "solves" the problem until it is fixed upstream.

**Comment by Legion2** [comment]
I updated the quarkus version so the fix for quarkusio/quarkus#7404 is now included.
